pkgs/development/compilers/v/default.nix

```diff
@@ -0,0 +1,43 @@
+{ stdenv, fetchurl, fetchFromGitHub, glfw, freetype, curl }:
+
+let
+    # V compiler's source translated to C for bootstrapping.
+    vc = fetchurl {
+        url = "https://github.com/vlang/v/releases/download/v0.1.0/v.c";
+        sha256 = "0vm5f0538c4gflj3k0kpin2f3iw7r4xi75cmiincz1913d1hbkh7";
+    };
+in stdenv.mkDerivation rec {
+    name = "v-${version}";
+    version = "0.1.0";
+
+    src = fetchFromGitHub {
+        owner = "vlang";
+        repo = "v";
+        rev = "v${version}";
+        sha256 = "0jdzlgmrkvwwnwsnd8m5zsqyagxiaval45q1mqqdrxhwsg54b8b3";
+    };
+
+    buildInputs = [ glfw freetype curl ];
+
+    buildPhase = ''
+        export HOME=`pwd`
+        mkdir -p code/v
+        cp -r `ls -A | grep -v "code"` code/v
```

Comment:
Do you need to create a new directory? Their README recommends to build in the `compiler` subdirectory. You could use `sourceRoot` to go there.
Especially this line looks  quite complicated at first glance (If you use `mv` you don't need `grep -v`, I think)

Reply:
Yes, home must be set, and there must be a code directory, in which the V directory goes that contains the files and sub-directories.
See: https://github.com/vlang/v/issues/457
That `cp` command copies everything but the `code` directory, is there a better command for this?

Reply:
No need for code directory anymore.